doc/source/whatsnew/v2.0.0.rst

```diff
@@ -288,9 +288,11 @@ Other enhancements
 - :meth:`Series.dropna` and :meth:`DataFrame.dropna` has gained ``ignore_index`` keyword to reset index (:issue:`31725`)
 - Improved error message in :func:`to_datetime` for non-ISO8601 formats, informing users about the position of the first error (:issue:`50361`)
 - Improved error message when trying to align :class:`DataFrame` objects (for example, in :func:`DataFrame.compare`) to clarify that "identically labelled" refers to both index and columns (:issue:`50083`)
+- Performance improvement in :func:`to_datetime` when format is given or can be inferred (:issue:`50465`)
```

Comment:
We are in the processing of cutting 2.0.0 now; at this point should target 2.1.0 for this PR

Reply:
@kostyafarber, hi! Can you merge main in [issue-50395](https://github.com/kostyafarber/pandas/tree/issue-50395)?

Reply:
Yep will do

Reply:
@dimastbk merged!

Reply:
Looks like this is still in the v2.0.0 whatsnew

Reply:
Yes, I asked for merging main in pr branch. Now fixed.

Reply:
merged.

Reply:
You should revert any changes to this file - the v2.0.0.rst file shouldn't be touched as part of this PR